plugin/openassetio_manager_bal/BasicAssetLibraryInterface.py

```diff
@@ -189,6 +189,28 @@ def register(
                 )
                 successCallback(idx, self.__build_entity_ref(updated_entity_info))
 
+    def getRelatedReferences(
+        self, entityRefs, relationshipTraitsDatas, context, hostSession, resultTraitSet=None
+    ):
+        # Ensure we have equal length arrays by expanding singular values
+        # (the inputs will always be arrays, just may have one entry).
+        # The middleware takes care of assuring that the are already
+        # equal in length if not either of the 1:many cases.
+        if len(entityRefs) == 1 and len(relationshipTraitsDatas) > 1:
+            entityRefs = entityRefs * len(relationshipTraitsDatas)
```

Comment:
Oh cool, thanks. Hadn't seen that. What do you think to [this version](https://github.com/OpenAssetIO/OpenAssetIO-Manager-BAL/pull/18/commits/034fa7af936b62599b96df79cc2aa46aef5f9cb8)? It is also a bit "sneaky" (in the way the `fill_value` is dynamic), is it easier to grok?

Reply:
I think that's good. In fact, I know it is because I didn't fully understand what was happening until seeing that formulation just now, thought we were putting in default (ie, Null) values as the pair for the longer list, but now it's clear that we're adding duplicates of the singular value to turn 1:X into X:X.